art/attacks/poisoning/hidden_trigger_backdoor/hidden_trigger_backdoor_pytorch.py

```diff
@@ -0,0 +1,313 @@
+# MIT License
+#
+# Copyright (c) 2020 Aniruddha Saha, Akshayvarun Subramanya, Hamed Pirsiavash
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements a Hidden Trigger Backdoor attack on Neural Networks.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+from typing import List, Optional, Tuple, Union, TYPE_CHECKING
+
+import numpy as np
+from tqdm.auto import trange
+
+from art.attacks.attack import PoisoningAttackWhiteBox
+from art.attacks.poisoning.backdoor_attack import PoisoningAttackBackdoor
+from art.estimators import BaseEstimator, NeuralNetworkMixin
+from art.estimators.classification.classifier import ClassifierMixin
+
+if TYPE_CHECKING:
+    # pylint: disable=C0412
+    from art.estimators.classification.pytorch import PyTorchClassifier
+
+logger = logging.getLogger(__name__)
+
+
+class LossMeter:
```

Comment:
Is this `LossMeter` the same as the one in `hidden_trigger_backdoor_keras.py`? Would it be possible to implement it once and reuse?

Reply:
Yeah, it's the same one across both. I originally tried to import it from `hidden_trigger_backdoor.py`, but I got a circular dependency. I could import it from `hidden_trigger_backdoor_keras.py`, but that might look odd. Maybe I should create it in a new python file and import from there?